android/src/main/java/com/adyenreactnativesdk/component/googlepay/AdyenGooglePayComponent.kt

```diff
@@ -109,12 +114,22 @@ class AdyenGooglePayComponent(context: ReactApplicationContext?) : BaseModule(co
 
     @ReactMethod
     fun hide(success: Boolean?, message: ReadableMap?) {
-        googlePayComponent = null
+        appCompatActivity.runOnUiThread {
+            loadingDialogFragment?.let {
+                googlePayComponent?.removeObservers(it)
+                googlePayComponent?.removeErrorObservers(it)
+            }
+            loadingDialogFragment?.dismiss()
+            loadingDialogFragment = null
+            googlePayComponent = null
+        }
         shared = null
     }
 
     private fun onError(error: Exception) {
-        sendErrorEvent(error)
+        Log.d(TAG, error.toString())
+        val componentError = if ((error as ComponentException?)?.message == "Payment canceled.") BaseModuleException.Canceled() else error
```

Comment:
for casting nullables you should use `error as? ComponentException` because it casts safely. also for convention :)

Reply:
Fixed here [c9b4c1b](https://github.com/Adyen/adyen-react-native/pull/57/commits/c9b4c1b955830b8a7056ca97450280d5a08154e1)